libs/fluid-elements/tab-group/src/lib/tab-group.ts

```diff
@@ -51,50 +52,77 @@ export class FluidTabGroup extends LitElement {
   }
 
   /**
-   * Defines a tab to be active
+   * Defines the currently active tab id
    * @attr
    * @type string
    */
   @property({ type: String, reflect: true })
-  activetabid: string;
+  activeTabId: string;
+
+  /** Contains the tab id that is focus using ARROWKEYS and will be activated when the user presses SPACEBAR */
+  private _toBeActiveTab: string;
 
   /** Sets the active tab on click */
   private handleClick(event: FluidTabActivatedEvent): void {
-    if (this.activetabid != event.activeTab) {
-      this.activetabid = event.activeTab;
-      for (const tab of this.tabChildren) {
-        tab.active = tab.tabid === this.activetabid;
+    const toActivateTab = this.tabChildren.find(
+      (tabItem) => tabItem.tabid === event.activeTab,
+    );
+
+    if (toActivateTab) {
+      // Only continue when the clicked tab is different from the currently active tab
+      if (this.activeTabId !== event.activeTab) {
```

Comment:
The ifs can be merged. 
```suggestion
    // Only continue when the clicked tab is different from the currently active tab
    if (toActivateTab && this.activeTabId !== event.activeTab) {
```